src/main/java/org/opensearch/knn/index/KNNMethodContext.java

```diff
@@ -41,7 +41,7 @@
  * KNNMethodContext will contain the information necessary to produce a library index from an Opensearch mapping.
  * It will encompass all parameters necessary to build the index.
  */
-@AllArgsConstructor
+@RequiredArgsConstructor
```

Comment:
Why switch?

Reply:
I switched it when I was adding another attribute here. Its not required. I will fix it.